tests/src/Kernel/Migrate/MigratePhoneNumberTestTrait.php

```diff
@@ -0,0 +1,201 @@
+    $this->assertNotNull($entity_form_display->getComponent('phone_number'));
+
+    // Rollback migration and check that verifications, phone number settings
+    // and phone number fields are removed.
+    $this->rollBackMigrations($this->getMigrationsToRollBack());
+
+    // Assert no phone number verifications, phone number settings or phone
+    // number fields exist.
+    $this->assertEquals([], PhoneNumberVerification::loadMultiple());
+    $this->assertEquals([], PhoneNumberSettings::loadMultiple());
+    $this->assertNull(FieldConfig::loadByName('user', 'user', 'phone_number'));
+    $this->assertNull(FieldStorageConfig::loadByName('user', 'phone_number'));
+
+    // Test that the display field is removed.
+    $entity_form_display = EntityFormDisplay::load('user.user.default');
+    $this->assertNull($entity_form_display->getComponent('phone_number'));
+  }
+
+  /**
+   * Asserts that the specified user has a verified phone number.
+   */
+  protected function assertVerifiedPhoneNumber(UserInterface $user, $number) {
+    $phone_numbers = $this->container->get('sms.phone_number')->getPhoneNumbers($user, TRUE);
+    $phone_number = reset($phone_numbers);
+    return $this->assertEquals($phone_number, $number, "Phone number '$number' is verified.");
+  }
+
+  /**
+   * Asserts that the specified user has an unverified phone number.
+   */
+  protected function assertUnVerifiedPhoneNumber(UserInterface $user, $number) {
+    $phone_numbers = $this->container->get('sms.phone_number')->getPhoneNumbers($user, FALSE);
+    $phone_number = reset($phone_numbers);
+    return $this->assertEquals($phone_number, $number, "Phone number '$number' is unverified.");
+  }
+
+  /**
+   * Asserts that the specified user has no phone number verified or unverified.
+   */
+  protected function assertNoVerifiedPhoneNumber(UserInterface $user) {
+    $phone_numbers = $this->container->get('sms.phone_number')->getPhoneNumbers($user);
+    return $this->assertEquals([], $phone_numbers, "No phone numbers for user {$user->id()}.");
+  }
+
+  /**
+   * Asserts that the specified number has a pending verification code.
+   */
+  protected function assertVerificationCode($number, $code) {
+    $verification = $this->container->get('sms.phone_number.verification')->getPhoneVerificationByPhoneNumber($number, FALSE);
+    $verification = reset($verification);
+    return $this->assertEquals($code, $verification->getCode());
+  }
+
+  /**
+   * Rolls back a specified migration.
+   */
+  protected function rollBackMigrations(array $ids) {
+    foreach ($ids as $id) {
+      $this->migration = $this->getMigration($id);
+      $this->prepareMigration($this->migration);
+      (new MigrateExecutable($this->migration, $this))->rollback();
+    }
+  }
+
+  /**
+   * Implementing classes must implement this to provide database query paths.
+   */
+  abstract protected function confirmationMessageFixturePath();
```

Comment:
Doc needs improvement. Should be more descriptive of what the query paths are / is for.